docs/explanation/cryptography.rst

```diff
@@ -0,0 +1,34 @@
+.. _explanation_cryptographic-technology:
+
+Cryptographic technology in Charmcraft
+======================================
+
+Charmcraft uses cryptographic technologies to fetch arbitrary files over the internet,
+communicate with local processes, and store credentials. It does not directly implement
+its own cryptography, but it does depend on external libraries to do so.
+
+Charmcraft is built upon `Craft Application`_ and derives much of its functionality from
+it, so much of Charmcraft's cryptographic functionality is described in `Cryptographic
+technology in Craft Application`_. The additional cryptography functionality in
+Charmcraft is documented below.
+
+Local Docker images
+~~~~~~~~~~~~~~~~~~~
+
+Charmcraft uses the `Docker SDK for Python`_ to communicate with a local Docker instance
+(if one is installed or running). This communicates with Docker over a Unix-domain
+socket if the user running Charmcraft has the necessary permission.
```

Comment:
> installed or running on the host

I would rather we explain what the container could be doing and when Charmcraft would need to communicate with it.

Reply:
Sublime!